src/proposals.rs

```diff
@@ -123,9 +123,33 @@ async fn poll_approved_proposals(
 ) -> Result<(), Error> {
     let config = APP.config();
     let mut gov_client = GovQueryClient::connect(config.cosmos.grpc.clone()).await?;
+
+    // Tracks which pending proposals have been submitted already for skipping
+    let mut pending_already_processed = HashSet::<u64>::default();
+
+    // Used to track what ID to process when there are pending proposals
+    let mut last_checked_pending_id = 0;
+
+    // Whether there are pending proposals to check
+    let mut pending_proposals = false;
```

Comment:
I think we've overcomplicated this a bit by adding a state indicating pending proposals. Because you have the `HashSet` above, you can always record when you've submitted any "pending" proposals. The only state we need to increment is the `last_processed_proposal_id`, we just should not increment it passed anything that could be considered pending. If the name no longer makes sense, maybe `last_finalized_proposal_id` makes more sense. We can always re-process pending proposals because the `HashSet` prevents duplication, so we can just always start from the `last_processed_proposal_id` on a new check.

Reply:
I was imagining a scenario where a later pending ID is submitted before an earlier pending ID, in such a case we wouldn't want to move the last-processed state beyond the ID that hasn't been submitted. This could happen if iD 100 is stuck in deposit period (but will eventually be voted on) and ID 101 is deposited in full and voted on first.

Reply:
We don't need to care about the prop ID order in that sense -- the way the chain operates is that gov proposals should be fulfilled when they are approved. If someone is sequencing multiple things they want to happen in a certain order they have to account for that either with proposal times or the parameters they are using (e.g. the block execution height).

Reply:
What I meant was steward will stop processing IDs before it, but now that I think about it I don't think that's true because we have the separate last-pending state. Will fix

Reply:
It's fine to submit a later ID, because we can just inspect the hashset to see if we've already looked at it. We should only increment the prop ID state when it's the next sequential number (or as part of a series of missing prop IDs).

Reply:
Ok we're in agreement there, I was misreading your original comment. You're saying don't use a separate ID state list for pending proposals, just always start for the state we have, and only increment when the next sequential ID has been processed. Is that right?

Reply:
Yes, exactly. This way we can "skip" pending states to process completed props, but then we'll end up reprocessing them later to close them out.